connector_prestashop/unit/importer.py

```diff
@@ -492,7 +493,12 @@ def _after_import(self, binding):
 def import_batch(session, model_name, backend_id, filters=None, **kwargs):
     """ Prepare a batch import of records from PrestaShop """
     backend = session.env['prestashop.backend'].browse(backend_id)
-    env = backend.get_environment(model_name, session=session)
+    connector_env_options = kwargs.get('connector_env_options', False)
+    if connector_env_options:
+        env = backend.get_environment(
+            model_name, session=session, **connector_env_options)
+    else:
+        env = backend.get_environment(model_name, session=session)
```

Comment:
But in any `ConnectorUnit` you have access to the value stored in the `ConnectorEnvironment` so I don't get why you want to add **kwargs everywhere, why not propagate the `connector_env_options` as well in the jobs?. We should clearly defines what goes in and what goes out here and it seems that's just not clear at all actually.

No, you don't need to modify the core `ConnectorEnviroment`, you can make a Prestashop special one.